c++ warnings ignoring return value

https://cran.r-project.org/web/checks/check_results_RMixtCompIO.html

```
Version: 4.0.9
Check: whether package can be installed
Result: WARN
    Found the following significant warnings:
     RGraph.cpp:95:58: warning: ignoring return value of 'std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(__cxx11::basic_string<_CharT, _Traits, _Allocator>&&, const __cxx11::basic_string<_CharT, _Traits, _Allocator>&) [with _CharT = char; _Traits = char_traits<char>; _Alloc = allocator<char>]', declared with attribute 'nodiscard' [-Wunused-result]
     RGraph.h:122:41: warning: ignoring return value of 'std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(__cxx11::basic_string<_CharT, _Traits, _Allocator>&&, const __cxx11::basic_string<_CharT, _Traits, _Allocator>&) [with _CharT = char; _Traits = char_traits<char>; _Alloc = allocator<char>]', declared with attribute 'nodiscard' [-Wunused-result]
    See ‘/data/gannet/ripley/R/packages/tests-devel/RMixtCompIO.Rcheck/00install.out’ for details.
    * used C++ compiler: ‘g++-13 (GCC) 13.1.0’
```
Flavor: [r-devel-linux-x86_64-fedora-gcc](https://www.r-project.org/nosvn/R.check/r-devel-linux-x86_64-fedora-gcc/RMixtCompIO-00check.html)

--- a/RMixtCompIO/DESCRIPTION
+++ b/RMixtCompIO/DESCRIPTION
@@ -1,8 +1,8 @@
 Package: RMixtCompIO
 Type: Package
 Title: Minimal Interface of the C++ 'MixtComp' Library for Mixture Models with Heterogeneous and (Partially) Missing Data
-Version: 4.0.9
-Date: 2023-02-11
+Version: 4.0.10
+Date: 2023-05-07
 Authors@R: c(person("Vincent", "Kubicki", role = "aut"),
              person("Christophe", "Biernacki", role = "aut"),
              person("Quentin", "Grimonprez", role = c("aut", "cre"), email = "quentingrim@yahoo.fr"),

--- a/RMixtCompIO/NEWS
+++ b/RMixtCompIO/NEWS
@@ -1,3 +1,7 @@
+v 4.0.10 2023-05-07
+- fix cpp warnings
+- update negative binomial test
+
 v 4.0.9 2023-02-11
 - lint
 - change c++ standard from c++11 to c++17
@@ -6,7 +10,6 @@ v 4.0.9 2023-02-11
 v 4.0.8 2022-10-18
 - correct bitwise-instead-of-logical warnings
 - remove blockcluster as suggested package
-
 
 v 4.0.7 2022-01-13
 - manage eigen 3.4.0

--- a/RMixtCompIO/README.md
+++ b/RMixtCompIO/README.md
@@ -57,7 +57,7 @@ R must be installed, along with the following packages from the CRAN: *RcppEigen
 Compiling R packages requires [Rtools](https://cran.r-project.org/bin/windows/Rtools/).
 
 
-## Build and install
+## Install
 
 Run in the current location (./RMixtCompIO/ folder):
 
@@ -68,11 +68,50 @@ make all
 
 The target *updateLib* copies the MixtComp library C++ folder from [MixtComp/src/lib](../MixtComp/src/lib) folder to [src/lib folder](src/lib).
 
+## Build
+
+```bash
+R CMD build RMixtCompIO
+```
+
+It generates a tar.gz file
+
+
+## Check
+
+```bash
+R CMD check --as-cran RMixtCompIO
+```
+
+or you can build the package before
+
+```bash
+R CMD check --as-cran RMixtCompIO_*.tar.gz
+```
+
+### Windows
+
+If you do not have windows, build the source and upload the tar.gz at <https://win-builder.r-project.org/upload.aspx>
+
+### Custom linux configuration
+
+To check on (almost) the same linux configuration as the CRAN, you can use the following procedure: <https://r-hub.github.io/containers/local.html>
+
+1) Install [Docker](https://docs.docker.com/engine/install/ubuntu/)
+1) Create a `check` folder and put the tar.gz package inside
+1) Run
+
+```bash
+docker run -v `pwd`/check:/check ghcr.io/r-hub/containers/gcc13:latest r-check
+```
+
+See <https://r-hub.github.io/containers/containers.html> for available containers (`clang16`, `clang-asan`, ...)
+
 ## Testing procedures
 
 ### Test the R package
 
-Tests are implemented in the `tests/testthat` folder of RMixtComp. They require the *testthat* package.
+Tests are implemented in the `tests/testthat` folder of RMixtCompIO. They require the *testthat* package.
 
 You can run tests via R:
 

--- a/RMixtCompIO/src/RGraph.cpp
+++ b/RMixtCompIO/src/RGraph.cpp
@@ -92,7 +92,7 @@ Rcpp::List RGraph::addSubGraph(const std::vector<std::string>& path, Index currD
 		} else if (TYPEOF(currLevel[path[currDepth]]) != VECSXP) { // if it already exists but is not a json object, throw an exception
 			std::string askedPath;
 			for (Index i = 0; i < currDepth + 1; ++i) {
-				askedPath + "/" + path[i];
+			  askedPath = askedPath + "/" + path[i];
 			}
 			throw(askedPath + " already exists and is not an R list.");
 		}

--- a/RMixtCompIO/src/RGraph.h
+++ b/RMixtCompIO/src/RGraph.h
@@ -119,7 +119,7 @@ private:
 			} else if (TYPEOF(currLevel[path[currDepth]]) != VECSXP) { // if it already exists but is not a json object, throw an exception
 				std::string askedPath;
 				for (Index i = 0; i < currDepth + 1; ++i) {
-					askedPath + "/" + path[i];
+				  askedPath = askedPath + "/" + path[i];
 				}
 				throw(askedPath + " already exists and is not an R list.");
 			}

--- a/RMixtCompIO/tests/testthat/test.run.R
+++ b/RMixtCompIO/tests/testthat/test.run.R
@@ -143,7 +143,7 @@ test_that("NegativeBinomial model works", {
   var$nBinom3 <- nBinomParamRandom("nBinom3")
   var$nBinom4 <- nBinomParamRandom("nBinom4")
 
-  resGen <- dataGeneratorNewIO(nInd, 0.9, var)
+  resGen <- dataGeneratorNewIO(nInd, 0.95, var)
 
   algo <- list(
     nClass = 2,

